docs/swarm.md

```diff
@@ -0,0 +1,81 @@
+<!--[metadata]>
++++
+title = "Using Compose with Swarm"
+description = "How to use Compose and Swarm together to deploy apps to multi-host clusters"
+keywords = ["documentation, docs,  docker, compose, orchestration, containers, swarm"]
+[menu.main]
+parent="workw_compose"
++++
+<![end-metadata]-->
+
+
+# Using Compose with Swarm
+
+Docker Compose and [Docker Swarm](/swarm/overview) aim to have full integration, meaning
+you can point a Compose app at a Swarm cluster and have it all just work as if
+you were using a single Docker host.
+
+The actual extent of integration depends on which version of the [Compose file
+format](compose-file.md#versioning) you are using:
+
+1.  If you're using version 1 along with `links`, your app is unlikely to work
+    out of the box unless all containers are scheduled on one host, because
+    links between containers do not work across hosts with the old networking
+    system.
+
+2. If you're using version 2, your app should work with no changes:
+
+    - subject to the [limitations](#limitations) described below,
+
+    - as long as the Swarm cluster is configured to use the [overlay
+      driver](/engine/userguide/networking/dockernetworks.md#an-overlay-network),
+      or a custom driver which supports multi-host networking.
+
+Read the [Getting started with multi-host
+networking](/engine/userguide/networking/get-started-overlay.md) to see how to
+set up a Swarm cluster with [Docker Machine](/machine/overview) and the overlay driver.
+Once you've got it running, deploying your app to it should be as simple as:
+
+    $ eval "$(docker-machine env --swarm <name of swarm master machine>)"
+    $ docker-compose up
+
+
+## Scheduling containers
+
+Swarm offers a rich set of scheduling and affinity hints, enabling you to
+control where containers are located. They are specified via container
+environment variables, so you can use Compose's `environment` option to set
+them.
+
+    environment:
+      # Schedule containers on a node that has the 'storage' label set to 'ssd'
+      - "constraint:storage==ssd"
+
+      # Schedule containers where the 'redis' image is already pulled
+      - "affinity:image==redis"
+
+For the full set of available filters and expressions, see the [Swarm
+documentation](/swarm/scheduler/filter/).
+
+
+## Limitations
```

Comment:
We might need to mention that `volumes_from` and `network_mode: service:x` still co-schedule containers, which breaks any applications that have dependencies like: `A -> C, B -> C`.